doc/plugins/authoring.rst

```diff
@@ -250,6 +250,29 @@ discover and load plugins. The glue which hooks things together is in
 NixOps implements a handful of hooks which your plugin can integrate
 with. See ``nixops/plugins/hookspec.py`` for a complete list.
 
+Defining a plugin
+-----------------
+
+To define a plugin you need to inherit from the `Plugin` base class and
+return it in the appropriate plugin hook:
+
+.. code-block:: python
+
+  from nixops.plugins import Plugin
+  import nixops.plugins
+
+  class MyPlugin(Plugin):
+
+      @staticmethod
+      def nixexprs():
+          return [ ]  # List of paths to nix expressions
```

Comment:
The doc above suggests changing some code to this format:

https://github.com/NixOS/nixops/blame/881c0b4baa90a57aa1be706beb1e897c6e8732bc/doc/plugins/authoring.rst#L94-L104

I think we should update those examples above, and then maybe include an example here of how to include a file.

Also, can we include a copy of the entire interface here somehow?